### PR TITLE
lowering dynamic shape split op in torch-blade

### LINKED ISSUE #1015: fix lowering failed if `aten.split`'s operand is dynamic shape
IR example lowering failed on DiscDecomposeComplexOps pass:

``` mlir
  %9 = torch.aten.reshape %7, %8 : !torch.vtensor<[?,?],f16>, !torch.list<int> -> !torch.vtensor<[?,?,?],f16>
  %10 = torch.operator "aten.split.Tensor"(%9, %int1280, %int-1) : (!torch.vtensor<[?,?,?],f16>, !torch.int, !torch.int) -> !torch.list<vtensor>
  %11 = torch.aten.__getitem__.t %10, %int0 : !torch.list<vtensor>, !torch.int -> !torch.vtensor<[?,?,?],f16>
  %12 = torch.aten.__getitem__.t %10, %int1 : !torch.list<vtensor>, !torch.int -> !torch.vtensor<[?,?,?],f16>
  %13 = torch.aten.to.dtype %12, %int6, %false, %false, %none : !torch.vtensor<[?,?,?],f16>, !torch.int, !torch.bool, !torch.bool, !torch.none -> !torch.vtensor
```

### PR DESCRIPTION
**Pull request by Yancey1989**
fixed #1015

### CODE REVIEW

File: pytorch_blade/pytorch_blade/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscSimplifyPatterns.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+#include "torch-mlir/Dialect/Torch/Utils/Utils.h"
+#include "torch-mlir/Dialect/TorchConversion/IR/TorchConversionOps.h"
+
+using namespace mlir;
+using namespace mlir::torch;
+using namespace mlir::torch::Torch;
+using namespace mlir::torch::TorchConversion;
+namespace {
+
+class SimplifyGetitemOp : public OpConversionPattern<Aten__Getitem__TOp> {
```
Comment (tanyokwok) [inline pytorch_blade/pytorch_blade/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscSimplifyPatterns.cpp:29]: Can Torch-MLIR simplify the getitems?

Comment (Yancey1989) [reply pytorch_blade/pytorch_blade/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscSimplifyPatterns.cpp:29]: I don't find a pass to simplifier this pattern yet... maybe torch-mlir can do that in the future, let's remove this if it exists.

File: pytorch_blade/pytorch_blade/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscDecomposeComplexOps.cpp [hunk 2]
```diff
@@ -84,16 +84,31 @@ class ConvertAtenOp : public OpConversionPattern<AtenOpT> {
       ConversionPatternRewriter& rewriter) const override;
 };
 
+int64_t getMaxIndexFromItemOps(OperatorOp op) {
```
[the remark this answers is not in the document]
Comment (Yancey1989) [reply pytorch_blade/pytorch_blade/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscDecomposeComplexOps.cpp]: done

File: pytorch_blade/pytorch_blade/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscSimplifyPatterns.cpp (hunk 1 above)
Comment (tanyokwok) [reply pytorch_blade/pytorch_blade/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscSimplifyPatterns.cpp:29]: Ok，I agree with you.

**Review by tanyokwok — APPROVED** [review]
LGTM

**Pull request merged**